data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -339,139 +424,61 @@ public InputStream executeStreamingQuery(String database, String command) throws
+                .newPost(clusterEndpoint)
+                .createCommandPayload(kr)
+                .withTracing(tracing)
+                .withAuthorization(authorization)
+                .build();
 
-        UUID activityId = UUID.randomUUID();
-        String activityContext = String.format("%s%s/%s, ActivityId=%s, ParentId=%s, ClientRequestId=%s", JAVA_INGEST_ACTIVITY_TYPE_PREFIX, activityTypeSuffix,
-                activityId, activityId, activityId, clientRequestId);
-        headers.put("x-ms-activitycontext", activityContext);
-
-        // replace non-ascii characters in header values with '?'
-        headers.replaceAll((_i, v) -> v == null ? null : v.replaceAll("[^\\x00-\\x7F]", "?"));
-        return headers;
-    }
-
-    Map<String, String> extractTracingHeaders(ClientRequestProperties properties) {
-        Map<String, String> headers = new HashMap<>();
-
-        String version = clientDetails.getClientVersionForTracing();
-        if (StringUtils.isNotBlank(version)) {
-            headers.put(CLIENT_VERSION_HEADER, version);
-        }
-
-        String app = (properties == null || properties.getApplication() == null) ? clientDetails.getApplicationForTracing() : properties.getApplication();
-        if (StringUtils.isNotBlank(app)) {
-            headers.put(APP_HEADER, app);
-        }
-
-        String user = (properties == null || properties.getUser() == null) ? clientDetails.getUserNameForTracing() : properties.getUser();
-        if (StringUtils.isNotBlank(user)) {
-            headers.put(USER_HEADER, user);
-        }
-
-        return headers;
+        // Get the response and trace the call
+        return MonitoredActivity.invoke(
+                (SupplierOneException<InputStream, DataServiceException>) () -> postToStreamingOutput(request),
+                "ClientImpl.executeStreamingQuery", updateAndGetExecuteTracingAttributes(kr.getDatabase(), kr.getProperties()));
     }
 
-    private String generateCommandPayload(String database, String command, ClientRequestProperties properties) {
-
-        ObjectNode json = objectMapper.createObjectNode()
-                .put("db", database)
-                .put("csl", command);
-
-        if (properties != null) {
-            json.put("properties", properties.toString());
+    private String getAuthorizationHeaderValue() throws DataServiceException, DataClientException {
+        if (aadAuthenticationHelper != null) {
+            return String.format("Bearer %s", aadAuthenticationHelper.acquireAccessToken());
         }
-
-        return json.toString();
-    }
-
-    private void addCommandHeaders(Map<String, String> headers) {
-        headers.put(HttpHeaders.CONTENT_TYPE, "application/json; charset=utf-8");
+        return null;
     }
 
     public String getClusterUrl() {
         return clusterUrl;
     }
 
+    ClientDetails getClientDetails() {
+        return clientDetails;
+    }
+
     @Override
     public void close() throws IOException {
```

Comment:
Do we still want it to be closeable in this case?

Reply:
fixed